### DISCUSSION TITLE
Neovim renders the local options of the current buffer in the other open buffers

### DISCUSSION THREAD

**Discussion by gmr458** [neovim/neovim#26095] [Q&A] [answered]
Hi, I was playing around trying to configure indentation lines without plugins using `vim.opt.list` and `vim.opt.listchars`, but I found that when there is two buffers open with different `shiftwidth` the `shiftwidth` of the current buffer is rendered in the other buffer.

Look at this example, the cursor is in the top buffer so that would make it our current buffer, I ran the command `:setlocal shiftwidth` so you can see that the current buffer has the local `shiftwidth` option set to 2. You can see how the bottom buffer appears to have `shiftwidth=2` when in fact it is set to 4.
![20231117_11h39m57s_grim](https://github.com/neovim/neovim/assets/53343401/1a71d43c-2f15-4204-8855-c5b8fa7c9a37)

When I switch to the bottom buffer (look at the next screenshot) you can see that it uses `shiftwidth=2`, and the same thing happens as before, but in reverse, the top buffer is rendered as if it used `shiftwidth=4`.
![20231117_11h44m02s_grim](https://github.com/neovim/neovim/assets/53343401/4cf0aba6-23c5-486e-b717-731859ae5826)

Is it supposed to work like that?
Is there a way to make each buffer continue rendering its local options?

**Comment by zeertzjq** [comment, marked as answer]
`vim.opt.listchars` sets the global value. To set the local value use `vim.opt_local.listchars`.

**Reply by gmr458** [reply]
It makes sense, I don't know why it didn't occur to me, thanks.
